velox/vector/ComplexVector.cpp

```diff
@@ -528,9 +528,23 @@ void ArrayVectorBase::copyRangesImpl(
 
 void RowVector::validate(const VectorValidateOptions& options) const {
   BaseVector::validate(options);
+  std::optional<vector_size_t> lastSetBit;
+
+  if (nulls_) {
+    lastSetBit = bits::findLastBit(nulls_->as<uint64_t>(), 0, size());
+  }
+
   for (auto& child : children_) {
     if (child != nullptr) {
       child->validate(options);
+      if (child->size() < size()) {
+        // Cannot have null nulls buffer if child size < parent size.
+        VELOX_CHECK_NOT_NULL(nulls_);
```

Comment:
nit: add a message at the end explaining why this is required, then you dont need the comment and the log will have a helpful message. Same for L546